sys/include/string_utils.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2022 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    sys_string_utils    Utility functions that are missing in
+ *                                  `string.h`
+ * @ingroup     sys
+ *
+ * This header provides utility functions that the standard C libs `string.h`
+ * lacks, such as @ref explicit_bzero
+ *
+ * @{
+ *
+ * @file
+ * @brief       Utility functions that are missing in `string.h`
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ */
+
+#include <string.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#ifndef STRING_UTILS_H
+#define STRING_UTILS_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef HAS_EXPLICIT_BZERO
```

Comment:
Do I see this right that you're waiting for Murdock to figure out on which boards to set this?

Reply:
It will actually depend on the C lib. It could be, that this worka as is everywhere, as newer versions of newlib do provide it, but as symbol and the declaration is guarded by preprocessor conditionals.

In any case it may be worth to either keep this here or a comment, so that people stumbling upon a compilation issue with conflicting decorations / implementations have an idea how to fix this.